opam-quickchik-1.0.2 install fails with coq 8.8.2

Due to other package dependencies (https://github.com/MetaCoq/metacoq/issues/92), I am stuck with coq-8.8.2 and I just discovered that QuickChick does not install. I tested and the package installs under coq-8.9.1.

``` $ opam install coq-quickchick                            11:00:24
The following actions will be performed:
  ∗ install coq-quickchick 1.0.2

<><> Gathering sources ><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
[coq-quickchick.1.0.2] found in cache

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
[ERROR] The compilation of coq-quickchick failed at "/usr/bin/make -j4".

#=== ERROR while compiling coq-quickchick.1.0.2 ===============================#
# context     2.0.4 | linux/x86_64 | base-bigarray.base base-threads.base base-unix.base ocaml-variants.4.07.0+flambda | https://coq.inria.fr/opam/released#2019-05-28 15:20
# path        ~/.opam/4.07.0+flambda/.opam-switch/build/coq-quickchick.1.0.2
# command     /usr/bin/make -j4
# exit-code   2
# env-file    ~/.opam/log/coq-quickchick-10087-05296d.env
# output-file ~/.opam/log/coq-quickchick-10087-05296d.out
### output ###
# [...]
# Hg' : semGenSize g s' (Some a)
# Hp' : p a
# Unable to unify "true" with "s' < 2 * (k + n.+1)".
# 
# Makefile.coq:656: recipe for target 'src/GenLow.vo' failed
# make[2]: *** [src/GenLow.vo] Error 1
# make[2]: *** Waiting for unfinished jobs....
# Makefile.coq:317: recipe for target 'all' failed
# make[1]: *** [all] Error 2
# make[1]: Leaving directory '/home/lord/.opam/4.07.0+flambda/.opam-switch/build/coq-quickchick.1.0.2'
# Makefile:24: recipe for target 'plugin' failed
# make: *** [plugin] Error 2

```

Attached env and out log files.

[coq-quickchick-10087-05296d.env.gz](https://github.com/coq/opam-coq-archive/files/3233838/coq-quickchick-10087-05296d.env.gz)
[coq-quickchick-10087-05296d.out.gz](https://github.com/coq/opam-coq-archive/files/3233839/coq-quickchick-10087-05296d.out.gz)

--- a/extra-dev/packages/coq-quickchick/coq-quickchick.8.8.dev/opam
+++ b/extra-dev/packages/coq-quickchick/coq-quickchick.8.8.dev/opam
@@ -14,7 +14,7 @@ depends: [
   "ocaml" {>= "4.04.0"}
   "coq" {>= "8.8" & < "8.9"}
   "coq-ext-lib"
-  "coq-mathcomp-ssreflect"
+  "coq-mathcomp-ssreflect" {< "1.9~"}
   "ocamlbuild"
   "ocamlfind"
   "coq-simple-io"

--- a/released/packages/coq-quickchick/coq-quickchick.1.0.1/opam
+++ b/released/packages/coq-quickchick/coq-quickchick.1.0.1/opam
@@ -14,7 +14,7 @@ depends: [
   "ocaml" {>= "4.04.0"}
   "coq" {>= "8.8" & < "8.9~"}
   "coq-ext-lib"
-  "coq-mathcomp-ssreflect"
+  "coq-mathcomp-ssreflect" {< "1.9~"}
   "ocamlbuild"
   "ocamlfind"
 ]

--- a/released/packages/coq-quickchick/coq-quickchick.1.0.2/opam
+++ b/released/packages/coq-quickchick/coq-quickchick.1.0.2/opam
@@ -14,7 +14,7 @@ depends: [
   "ocaml" {>= "4.04.0"}
   "coq" {>= "8.8" < "8.9~"}
   "coq-ext-lib"
-  "coq-mathcomp-ssreflect"
+  "coq-mathcomp-ssreflect" {< "1.9~"}
   "ocamlbuild"
   "ocamlfind"
   "coq-simple-io" {>= "0.2" & < "0.3"}

